File: python/leetcode/algorithms/medium/442-数组中重复的数据.py
from typing import List


class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        res = []
        for i, num in enumerate(nums):
            newNum = abs(num) - 1
            if nums[newNum] < 0:
                res.append(abs(num))
            else:
                nums[newNum] = -nums[newNum]
        return res


nums = [4, 3, 2, 7, 8, 2, 3, 1]
s = Solution()
print(s.findDuplicates(nums))

File: go/leetcode/medium/442-数组中重复的数据.go
package main

import "math"

func findDuplicates(nums []int) []int {
	var result []int
	for _, num := range nums {
		newNum := int(math.Abs(float64(num)))
		if nums[newNum-1] < 0 {
			result = append(result, newNum)
		} else {
			nums[newNum-1] = -nums[newNum-1]
		}
	}
	return result
}

func main() {

}
